apps/communications/contacts/js/activities_vcard.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+/* global LazyLoader, VCardReader, Contacts */
+/* exported vCardActivityHandler */
+
+var vCardActivityHandler = function() {
+  var handle = (activity, parentHandler) => {
+    const DEPENDENCIES = '/shared/js/contacts/import/utilities/vcard_reader.js';
+    LazyLoader.load(DEPENDENCIES, function () {
+      getvCardReader(activity.source.data.blob).then(readvCard, (err) => {
+        console.error(err);
+        Contacts.cancel();
+      });
+    });
+
+    function getvCardReader(blob) {
+      return new Promise((resolve, reject) => {
+        var fileReader = new FileReader();
+        fileReader.readAsText(blob);
+        fileReader.onloadend = () => {
+          resolve(new VCardReader(fileReader.result));
+        };
+        fileReader.onerror = () => {
+          reject('The blob could not be read as a file.');
+        };
+      });
+    }
+
+    function readvCard(vCardReader) {
+      var firstContact;
+      var cursor = vCardReader.getAll();
+      cursor.onsuccess = function(event) {
+        var contact = event.target.result;
+        // We check if there is only one contact to know what function
+        // we should call. If not, we render the contacts one by one.
+        if (contact) {
+          if (!firstContact) {
+            firstContact = contact;
+            cursor.continue();
+          } else {
+            renderContacts([firstContact, contact], activity, cursor);
+          }
+        } else if (firstContact) {
+          renderOneContact(firstContact, activity);
+        }
+      };
+    }
+
+    function renderOneContact(contact, activity) {
+      parentHandler.mozContactParam = contact;
+      var data = activity.source.data;
+      data.params = {'mozContactParam': true};
+      if (data.allowSave === true) {
+        parentHandler.launch_activity(activity, 'view-contact-form');
+      } else {
+        parentHandler.launch_activity(activity, 'view-contact-details');
+      }
+    }
+
+    function renderContacts(contacts, activity, cursor) {
+      parentHandler.launch_activity(activity, 'multiple-select-view');
+      const DEPENDENCIES = [
+        '/contacts/js/views/multiple_select.js'
+      ];
+      LazyLoader.load(DEPENDENCIES, () => {
+        var filename = activity.source.data.filename;
+        Contacts.MultipleSelect.init();
```

Comment:
couldn't this be something internal to the MultipleSelect component i.e. the 'MultipleSelect' component just calls itself its 'init' method?